lib/Server.js

```diff
@@ -251,23 +251,29 @@ Server.prototype.routes.deleteBucketToken = function(req, res, done) {
 
 Server.prototype.routes.deleteAssetFromBucket = function(req, res, done) {
   var self = this;
-  var bucket = req.params.bucket;
+  var bucketId = req.params.bucket;
   var assetName = req.params.assetName;
-
-  this.database.deleteBucketAsset(bucket, assetName, function(error, data) {
-    if (error) {
-      return res.status(500).send('Asset could not be deleted.');
-    }
-    self.fileStorage.deleteFile(data, function(err) {
-      if (err) {
-        return res.status(500).send('Asset could not be deleted.');
-      }
-      else {
-        return res.status(202).send('Asset removed.');
-      }
-    });
+  var flush = function(cb) {
+    res.writeHead(202);
+    res.end('Asset deleted');
+    cb();
+    done();
+  };
+  this.database.deleteBucketAsset(bucketId, assetName, function(err) {
+    if (err) { return self.handleError(req, res, err); }
+    self.database.getAssetVersionsByAssetId(bucketId, assetName)
+      .pipe(through2.obj(function(assetId, enc, cb) {
+        this.push(assetId);
+        self.database.deleteBucketAssetVersion(bucketId, assetName, assetId, function(err) {
+          if (err) { return self.handleError(req, res, err); }
+          self.fileStorage.deleteFile(assetId, function(err) {
+            if (err) { self.handleError(req, res, err); }
+            self.logger.info(`Deleted asset file: ${assetId}`);
```

Comment:
We should always be logging structured content so we should probably have `{bucketId, assetId}` here.